test/dataclasses/test_tool.py

```diff
@@ -129,6 +129,35 @@ def function_with_docstring(city: str) -> str:
         }
         assert another_tool.function == function_with_docstring
 
+    def test_from_function_with_custom_description(self):
+        another_tool = Tool.from_function(
+            function=function_with_docstring,
+            description="custom description",
+        )
+
+        assert another_tool.name == "function_with_docstring"
+        assert another_tool.description == "custom description"
+        assert another_tool.parameters == {
+            "type": "object",
+            "properties": {"city": {"type": "string"}},
+            "required": ["city"],
+        }
+        assert another_tool.function == function_with_docstring
+
+    def test_from_function_name_from_function(self):
```

Comment:
`test_from_function_with_custom_name` would be more in line with the other tests `test_from_function_with_custom_description` and `test_from_function_with_empty_description`. And 
```python
        another_tool = Tool.from_function(
            function=function_with_docstring,
            name="custom_name",
        )

        assert another_tool.name == "custom_name"
```
would be enough to test in this test case. The other case, when there is no custom name provided, is already being tested as part of the other test cases.

Reply:
Makes sense. Done!